### DISCUSSION TITLE
Why does "each" group the first two lines here?

### DISCUSSION THREAD

**Discussion by MatrixManAtYrService** [nushell/nushell#12131] [Q&A] [answered]
I was hoping to switch from using jq to using nushell in situations like this:

```nushell
 ❯ python -c '
∙ from time import sleep
∙ from json import dumps
∙ from sys import stdout, stderr
∙ def f(s):
∙     print(dumps({"key":s}))
∙     stdout.flush()
∙     sleep(1)
∙ for i, s in enumerate(["foo", "bar", "baz"]):
∙     print(i, file=stderr)
∙     f(s)
∙ ' | jq 'select(.key == "foo")'
    0
    {
      "key": "foo"
    }
    1
    2
```

`each` seemed like a good candidate, so I did this:

```nushell
 ❯ : python -c '
∙ from time import sleep
∙ from json import dumps
∙ from sys import stdout, stderr
∙ def f(s):
∙     print(dumps({"key":s}))
∙     stdout.flush()
∙     sleep(1)
∙ for i, s in enumerate(["foo", "bar", "baz"]):
∙     print(i, file=stderr)
∙     f(s)
∙ ' | each {|l| $l | from json}
    0
    1
    ╭───┬─────╮
    │ # │ key │
    ├───┼─────┤
    │ 0 │ foo │
    │ 1 │ bar │
    ╰───┴─────╯
     2
    ╭───┬─────╮
    │ # │ key │
    ├───┼─────┤
    │ 2 │ baz │
    ╰───┴─────╯
```

Why does foo end up grouped with bar here?  [This discussion](https://github.com/nushell/nushell/discussions/5521) seems relevant, but it doesn't quite get me where I'd like to go.

**Comment by fdncred** [comment, marked as answer]
nushell creates tables that are separated because of two things.
1. there are more than 1000 rows
2. the timeout has occured. this is so we don't have half tables hanging around waiting for input. i believe the timeout is 1 second.

If you want to avoid this, you can collect at the end of your pipeline, but then it's not streaming.
```nushell
| each {|l| $l | from json} | collect {$in}
```
